# **Exploratory Data Analysis (EDA)**

## Descriptive statistics for numeric columns

In [7]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
df = pd.read_csv("processed_data.csv")
df.head()

Video ID                                              Title  Duration  \
0  KbiwL74KyJQ                          Shrek 5 Cast Announcement        28   
1  8B1EtVPBSMw                  A Minecraft Movie | Final Trailer       151   
2  W7FTkUFU7nw                       Pokémon Presents | 2.27.2025      1156   
3  r5VRqWkFpEQ  LISA - FUTW (Vixi Solo Version) (Official Musi...       228   
4  vONxgCQWZCA  YoungBoy Never Broke Again - 5 Night [Official...       196   

      Views     Likes  Comments           Publication Time Region  \
0  0.053542  0.034124  0.000000  2025-02-27 16:01:18+00:00     US   
1  0.080644  0.052269  0.132190  2025-02-27 20:00:04+00:00     US   
2  0.024396  0.034106  0.106517  2025-02-27 14:00:06+00:00     US   
3  0.055906  0.221836  0.382255  2025-02-28 05:01:26+00:00     US   
4  0.003523  0.013293  0.033900  2025-02-28 06:32:53+00:00     US   

   Topic_Autos & Vehicles  Topic_Comedy  ...  Language_Arabic  \
0                       0             0  ...                0   
1                       0             0  ...                0   
2                       0             0  ...                0   
3                       0             0  ...                0   
4                       0             0  ...                0   

   Language_English  Language_French  Language_German  Language_Hindi  \
0                 1                0                0               0   
1                 1                0                0               0   
2                 1                0                0               0   
3                 1                0                0               0   
4                 1                0                0               0   

   Language_Japanese  Language_Korean  Language_Portuguese  Language_Russian  \
0                  0                0                    0                 0   
1                  0                0                    0                 0   
2                  0                0                    0                 0   
3                  0                0                    0                 0   
4                  0                0                    0                 0   

   Language_Spanish  
0                 0  
1                 0  
2                 0  
3                 0  
4                 0  

[5 rows x 33 columns]

In [8]:
df.shape

(2510, 33)

based on above output, we have 2510 rows, where each row represent a video, and we have 33 columns.

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2510 entries, 0 to 2509
Data columns (total 33 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Video ID                     2510 non-null   object 
 1   Title                        2510 non-null   object 
 2   Duration                     2510 non-null   int64  
 3   Views                        2510 non-null   float64
 4   Likes                        2510 non-null   float64
 5   Comments                     2510 non-null   float64
 6   Publication Time             2510 non-null   object 
 7   Region                       2510 non-null   object 
 8   Topic_Autos & Vehicles       2510 non-null   int64  
 9   Topic_Comedy                 2510 non-null   int64  
 10  Topic_Education              2510 non-null   int64  
 11  Topic_Entertainment          2510 non-null   int64  
 12  Topic_Film & Animation       2510 non-null   int64  
 13  Topic_Gaming      